portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/PortalSiteHelperImpl.java

```diff
@@ -855,6 +857,33 @@ else if (req.getAttribute("siteId")!=null){
 
 
 		String pagePopupUrl = Web.returnUrl(req, "/page/");
+
+		//SAK-29660 - Refresh tool in the LHS page menu
+		String toolUrlPrefix = ServerConfigurationService.getToolUrl();
+		Iterator iPt = pTools.iterator();
+		String resetActionUrl="";
+		String placementId="";
+		ToolConfiguration placement=null;
+		while (iPt.hasNext()) {
+			placement = (ToolConfiguration) iPt.next();
+			String pageResetUrl = Web.serverUrl(req)+ ServerConfigurationService.getString("toolPath")+ "/"+placement.getId()+"/?panel=Main";
+			resetActionUrl = PortalStringUtil.replaceFirst(pageResetUrl, toolUrlPrefix, toolUrlPrefix + "-reset");
+			placementId = placement.getId();
+			if(ToolUtils.isPortletPlacement(placement)){
+				resetActionUrl = Web.serverUrl(req)
+						+ ServerConfigurationService.getString("portalPath")
+						+ URLUtils.getSafePathInfo(req) + "?sakai.state.reset=true";
+				m.put("toolPlacementIDJS", "_self");
+				m.put("isPortletPlacement", Boolean.TRUE);
+			}
+			else {
+				m.put("toolPlacementIDJS",Web.escapeJavascript("Main"+ placementId));
+			}
+			if(placement.getPageOrder()==1) {
+				break;
+			}
```

Comment:
I'm not 100% sure this if-statement is needed to get the refresh working for the Home Tool, even if it has synoptic tools.